### PR TITLE
Add Import Tracking Plan docs [DOC-687]

### PR DESCRIPTION
**Pull request by forstisabella**
### Proposed changes
Added "Import Tracking Plan" docs

### Merge timing
December 11, 2023

### Related issues (optional)
#4629

### CODE REVIEW

File: src/protocols/tracking-plan/create.md [hunk 1]
```diff
@@ -87,6 +89,34 @@ You can define which traits you expect to see passed in Identify or Group calls
 
 It's best to keep traits optional because Identify and Group are often called and pass only _new or changed_ traits, because Segment's client-side libraries (analytics.js, Swift, Kotlin) cache traits in local storage. See the [Identify Best Practices](/docs/connections/spec/best-practices-identify#when-and-how-often-to-call-identify) to learn more.
 
+### Upload a Tracking Plan
+
+You can make changes to a Tracking Plan by uploading a CSV to replace the rules of your existing Tracking Plan.
```
Comment (ranand) [inline src/protocols/tracking-plan/create.md:94]: or create a new Tracking Plan as well.

File: src/protocols/tracking-plan/create.md [hunk 2]
```diff
@@ -87,6 +89,34 @@ You can define which traits you expect to see passed in Identify or Group calls
...
+
+> info "Tracking Plans with imported libraries cannot be changed using the Upload a Tracking Plan method"
+> If you have a Tracking Plan with imported libraries, you must make changes to your Tracking Plan in the Segment app.
+
+To update a Tracking Plan by uploading a CSV file: 
+1. Click **Protocols** in the left navigation bar. 
+2. On the row of the Tracking Plan you want to edit, open the contextual menu(...) and select **View Tracking Plan**.
+3. Select **Edit Tracking Plan**.
+4. Click the **Import...** button and select **From CSV**. 
+5. Download the Tracking Plan template CSV and fill in the template CSV file with your new Tracking Plan rules, or make changes to your [downloaded Tracking Plan](#download-a-tracking-plan).
```
Comment (ranand) [inline src/protocols/tracking-plan/create.md:104]: Alternatively, you can also export/download a tracking plan, make changes and upload them via "From CSV" option

File: src/protocols/tracking-plan/create.md [hunk 3]
```diff
@@ -87,6 +89,34 @@ You can define which traits you expect to see passed in Identify or Group calls
...
+6. Once you've filled in the provided template or made changes to your downloaded Tracking Plan, add your CSV file to the file uploader and click **Upload**.
+
+After uploading your CSV file, you are redirected to the Upload & Download History page while the upload is in progress. If the CSV upload fails due to issues with the content of your Tracking Plan, you can download the file `error_report.csv`, which provides you the error in the "Error Details" column of your CSV file. If the upload fails due to issues with the uploaded file itself (a file that is too large, has too many rows, or is not a CSV file, for example), you'll be able to view the error directly in the Reports column. 
+
+Common reasons a CSV upload may fail include: 
+- A filetype other than .csv
+- More than one file uploaded at a time
+- A file that is empty besides the header row
+- A CSV file that exceeds 100,000 rows
+- A CSV file with more than 20,000 rules
```
Comment (chenxzhang) [inline src/protocols/tracking-plan/create.md:114]: should be 2000 rules

File: src/protocols/tracking-plan/create.md [hunk 4]
```diff
@@ -87,6 +89,34 @@ You can define which traits you expect to see passed in Identify or Group calls
...
+1. Click **Protocols** in the left navigation bar. 
+2. On the row of the Tracking Plan you want to edit, open the contextual menu(...) and select **View Tracking Plan**.
+3. Select **Edit Tracking Plan**.
+4. Click the **Import...** button and select **From CSV**. 
+5. Download the Tracking Plan template CSV and fill in the template CSV file with your new Tracking Plan rules, or make changes to your [downloaded Tracking Plan](#download-a-tracking-plan).
+6. Once you've filled in the provided template or made changes to your downloaded Tracking Plan, add your CSV file to the file uploader and click **Upload**.
+
+After uploading your CSV file, you are redirected to the Upload & Download History page while the upload is in progress. If the CSV upload fails due to issues with the content of your Tracking Plan, you can download the file `error_report.csv`, which provides you the error in the "Error Details" column of your CSV file. If the upload fails due to issues with the uploaded file itself (a file that is too large, has too many rows, or is not a CSV file, for example), you'll be able to view the error directly in the Reports column.
```
Comment (ranand) [inline src/protocols/tracking-plan/create.md:107]: @chenxzhang - do we wanna capture common cases in which we generate error_report.csv and cases when we show the error in the upload ui itself?

File: src/protocols/tracking-plan/create.md (hunk 3 above)
Comment (forstisabella) [reply src/protocols/tracking-plan/create.md:114]: good catch!

**Review by pwseg — APPROVED** [review]

File: src/protocols/tracking-plan/create.md (hunk 4 above)
Comment (chenxzhang) [reply src/protocols/tracking-plan/create.md]: Technically there are 3 types of errors: upload ui validation, upload & download history error for the general file itself, and upload & download history error report with specific error details.

1. upload ui validation (this happens as you upload the file from your local computer)
Common reasons for error:
- A filetype other than .csv
- An empty file
- A file that is empty besides the header row
- A CSV file that exceeds 100,000 rows
- A CSV file that does not have the expected number of columns/headers
- File name exceeding 80 characters

2.  upload & download history error for the general file itself 
- A CSV file that exceeds 2,000 rules
- A malformed CSV file (rows have different columns)
- Internal system errors

3. upload & download history that generates error_report
- Invalid label format (should be label:value)
- Invalid nested property (should be an object or array)

Comment (ranand) [reply src/protocols/tracking-plan/create.md]: cc: @forstisabella

Comment (forstisabella) [reply src/protocols/tracking-plan/create.md]: Thanks for the tag! 

I'll add a note under the "Upload a Tracking Plan" header with file size/formatting requirements, and make some changes to the troubleshooting section to slim it down.

Comment (forstisabella) [reply src/protocols/tracking-plan/create.md]: okay - I removed the "troubleshooting uploads" section and added a sentence after each subsection about the two ways users can see errors on the Upload & Download history page - is this a step in the right direction? @ranand

Comment (ranand) [reply src/protocols/tracking-plan/create.md]: Changes LGTM. Thanks for incorporating them!

Comment (forstisabella) [reply src/protocols/tracking-plan/create.md]: sweet!

**Review by chenxzhang — APPROVED** [review]

**Pull request merged**